drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -384,7 +384,7 @@ static int _set(netdev_t *netdev, netopt_t opt, void *val, size_t len)
             /* don't set res to set netdev_ieee802154_t::pan */
             break;
         case NETOPT_CHANNEL:
-            assert(len != sizeof(uint8_t));
+            assert(len <= sizeof(uint16_t));
```

Comment:
I don't get it. Why  should't I be allowed to pass a 8 bit channel if 802.15.4 channels always fit into 8 bit?